setup/gcp/installer/forseti_server_installer.py

```diff
@@ -0,0 +1,339 @@
+# Copyright 2017 The Forseti Security Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Forseti Server installer"""
+
+from __future__ import print_function
+import random
+
+from forseti_installer import ForsetiInstaller
+from util.utils import (
+    print_banner, format_resource_id, format_service_acct_id)
+from util.constants import (
+    MESSAGE_ENABLE_GSUITE_GROUP, MESSAGE_SETUP_IAM_EXPLAIN,
+    QUESTION_ENABLE_IAM_EXPLAIN, QUESTION_ACCESS_TO_GRANT_ROLES,
+    RESOURCE_TYPES, MESSAGE_FORSETI_CONFIGURATION_ACCESS_LEVEL,
+    QUESTION_FORSETI_CONFIGURATION_ACCESS_LEVEL, MESSAGE_ASK_SENDGRID_API_KEY,
+    QUESTION_SENDGRID_API_KEY, NOTIFICATION_SENDER_EMAIL, MESSAGE_SKIP_EMAIL,
+    QUESTION_NOTIFICATION_RECIPIENT_EMAIL, QUESTION_GSUITE_SUPERADMIN_EMAIL,
+    MESSAGE_ASK_GSUITE_SUPERADMIN_EMAIL, QUESTION_ENABLE_WRITE_ACCESS,
+    MESSAGE_HAS_ROLE_SCRIPT, MESSAGE_GSUITE_DATA_COLLECTION,
+    RULES_DIR_PATH, FirewallRuleAction, FirewallRuleDirection)
+from util.gcloud import (
+    enable_apis, create_reuse_service_acct, choose_organization,
+    choose_folder, choose_project, grant_server_svc_acct_roles,
+    create_firewall_rule)
+from util.files import copy_file_to_destination
+from configs.server_config import ServerConfig
+
+
+class ForsetiServerInstaller(ForsetiInstaller):
+    """Forseti server installer"""
+
+    # pylint: disable=too-many-instance-attributes
+    # Having ten variables is reasonable in this case.
+
+    gsuite_service_account = None
+    has_roles_script = False
+    setup_explain = True
+    enable_write_access = False
+    resource_root_id = None
+    access_target = None
+    target_id = None
+
+    def __init__(self, **kwargs):
+        """Init
+
+        Args:
+            kwargs (dict): The kwargs.
+        """
+        super(ForsetiServerInstaller, self).__init__()
+        self.config = ServerConfig(**kwargs)
+
+    def preflight_checks(self):
+        """Pre-flight checks for server instance"""
+
+        super(ForsetiServerInstaller, self).preflight_checks()
+
+        self.should_setup_explain()
```

Comment:
Explain setup should not be optional, right?  I thought it will be standard component in 2.0?

Reply:
Updated.